### DISCUSSION TITLE
Custom claim in JWT access token

### DISCUSSION THREAD

**Discussion by sagarshah1983** [ory/hydra#2552] [Q&A] [answered]
Hello everyone,
 I understand that we have capability to add custom claims in access token using consent flow as listed here - https://www.ory.sh/hydra/docs/concepts/consent#accepting-the-consent-flow

Is there any such thing possible for token endpoint using client_credentials flow as well? Intention is to add custom claim in token based on the some client's attribute (from metadata).

Appreciate your recommendations!

Regards,
Sagar

**Comment by aeneasr** [comment, marked as answer]
See https://github.com/ory/hydra/issues/1383 https://github.com/ory/hydra/issues/1748 :)

**Reply by sagarshah1983** [reply]
Thank you for providing links. So my understanding is "As of today, latest release of hydra does not support it, but it might be supported soon in newer release" I also understand that it might be harder to confirm on the target release, but if we can get rough timelines on when we can expect this, that will be great!
Appreciate your inputs

**Reply by aeneasr** [reply]
While I understand that a date of delivery is helpful when planning, we stopped giving out due dates for features or milestones. It puts maintainers in an unfair spot as

1. there are many things going on in the community which need time (reviewing PRs, addressing issues, answering questions) and demand is no not easily foreseeable;
2. maintainers do a lot of things in their free time and can not commit to deliver something;
3. internal priorization can shift at any point in time

When we do give out due dates, we often encounter people who are pissed off because something was promised and not delivered "on time" (as free software ;) ). Unfortunately that is the sad truth of many interactions in this context which is why we stopped doing it altogether.

Having said that, you are in the privileged position of being part in a global open source community! If you think that you can contribute towards the particular feature, maintainers will do their best to give you the right pointers and review PRs.

If that's not an option, you can consider to become a Ory Sponsor on [Open Collective](https://opencollective.com/ory) or [Patreon](https://www.patreon.com/_ory). This helps us to employ more maintainers and technical staff - increasing velocity in development, community, interaction, and other areas! All collected money goes directly into this.
